src/main/java/org/commcare/cases/util/StorageBackedTreeRoot.java

```diff
@@ -83,7 +85,18 @@ protected String translateFilterExpr(XPathPathExpr expressionTemplate, XPathPath
     private void collectPredicateProfiles(Vector<XPathExpression> predicates,
                                           Hashtable<XPathPathExpr, String> indices,
                                           EvaluationContext evalContext,
-                                          Vector<PredicateProfile> optimizations) {
+                                          Vector<org.commcare.cases.query.PredicateProfile> optimizations,
+                                          QueryContext queryContext) {
+
+        optimizations.addAll(getQueryPlanner().collectPredicateProfiles(predicates, queryContext, evalContext));
+
+        //For now we are going to skip looking deeper if we trigger
+        //any of the planned optimizations
```

Comment:
The old format for genreating predicateprofiles relied on being able to match things like @case_id on the left and some arbitrary value on the right that was generated by running the XPathExpression against the current evalContext. 

In cases where calculating the right hand side expression is the expensive part of the query this makes it ~impossible to optimize without preventing things from being generated.

In the future the best plan will be to delay the evaluation of the RightHandSide part of the predicate profile until it's ready for use in a specific handler, but I didn't have time to write/test that level of refactoring the process